server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/NettyBootstrap.scala

```diff
@@ -19,10 +20,20 @@ object NettyBootstrap {
       .group(eventLoopGroup)
       .channel(nettyConfig.eventLoopConfig.serverChannel)
       .childHandler(new ChannelInitializer[Channel] {
-        override def initChannel(ch: Channel): Unit = nettyConfig.initPipeline(nettyConfig)(ch.pipeline(), handler)
+        override def initChannel(ch: Channel): Unit = {
+          val nettyConfigBuilder = nettyConfig.initPipeline(nettyConfig)
+          nettyConfig.requestTimeout match {
+            case Some(requestTimeout) =>
+              nettyConfigBuilder(ch.pipeline().addLast(new ReadTimeoutHandler(requestTimeout.toSeconds.toInt)), handler)
+            case None => nettyConfigBuilder(ch.pipeline(), handler)
+          }
+        }
       })
       .option[java.lang.Integer](ChannelOption.SO_BACKLOG, nettyConfig.socketBacklog) // https://github.com/netty/netty/issues/1692
       .childOption[java.lang.Boolean](ChannelOption.SO_KEEPALIVE, nettyConfig.socketKeepAlive) // https://github.com/netty/netty/issues/1692
+      .childOption[java.lang.Integer](ChannelOption.SO_TIMEOUT, nettyConfig.socketTimeout.orNull.toSeconds.toInt)
```

Comment:
**issue:** This would throw a `NullPointerException` on `None`. You probably wanted something like `socketTimeout.map(_.toSeconds.toInt).orNull`. Are you sure that a `null` is OK here though?

Reply:
Yes, you're right, but @katlasik's solution resolves this problem ;) Besides, null is okay here, which means that I will remove this option if it exists, and if not, nothing will happen.